test-framework/amazon-lambda/src/main/java/io/quarkus/amazon/lambda/test/LambdaResourceManager.java

```diff
@@ -41,44 +40,46 @@ public void handleRequest(HttpServerExchange exchange) throws Exception {
             }
         });
-        routingHandler.add("POST", "/2018-06-01/runtime/invocation/{req}/response", new HttpHandler() {
-            @Override
-            public void handleRequest(HttpServerExchange exchange) throws Exception {
-                String id = exchange.getQueryParameters().get("req").getFirst();
-                exchange.getRequestReceiver().receiveFullString(new Receiver.FullStringCallback() {
+        routingHandler.add("POST", AmazonLambdaApi.API_PATH_INVOCATION + "{req}" + AmazonLambdaApi.API_PATH_RESPONSE,
+                new HttpHandler() {
                     @Override
-                    public void handle(HttpServerExchange exchange, String message) {
-                        LambdaClient.REQUESTS.get(id).complete(message);
+                    public void handleRequest(HttpServerExchange exchange) throws Exception {
+                        String id = exchange.getQueryParameters().get("req").getFirst();
+                        exchange.getRequestReceiver().receiveFullString(new Receiver.FullStringCallback() {
+                            @Override
+                            public void handle(HttpServerExchange exchange, String message) {
+                                LambdaClient.REQUESTS.get(id).complete(message);
+                            }
+                        });
                     }
                 });
-            }
-        });
 
-        routingHandler.add("POST", "/2018-06-01/runtime/invocation/{req}/error", new HttpHandler() {
-            @Override
-            public void handleRequest(HttpServerExchange exchange) throws Exception {
-                String id = exchange.getQueryParameters().get("req").getFirst();
-                exchange.getRequestReceiver().receiveFullString(new Receiver.FullStringCallback() {
+        routingHandler.add("POST", AmazonLambdaApi.API_PATH_INVOCATION + "{req}" + AmazonLambdaApi.API_PATH_ERROR,
+                new HttpHandler() {
                     @Override
-                    public void handle(HttpServerExchange exchange, String message) {
-                        ObjectMapper mapper = new ObjectMapper();
-                        try {
-                            FunctionError result = mapper.readerFor(FunctionError.class).readValue(message);
+                    public void handleRequest(HttpServerExchange exchange) throws Exception {
+                        String id = exchange.getQueryParameters().get("req").getFirst();
+                        exchange.getRequestReceiver().receiveFullString(new Receiver.FullStringCallback() {
+                            @Override
+                            public void handle(HttpServerExchange exchange, String message) {
+                                ObjectMapper mapper = new ObjectMapper();
+                                try {
+                                    FunctionError result = mapper.readerFor(FunctionError.class).readValue(message);
```

Comment:
I think we should cache the reader here. It can be done in another PR though as it was already preexisting. I don't know if we use this pattern elsewhere in the code.

Reply:
If it is safe to share a reader then +1 for a new PR ;-)

Reply:
Ah, I missed it's just in the tests. We don't really care then.